rules/no-object-as-default-parameter.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+const getDocumentationUrl = require('./utils/get-documentation-url');
+
+const MESSAGE_ID = 'noObjectAsDefaultParameter';
+
+const objectParameterSelector = [
+	':matches(FunctionDeclaration, ArrowFunctionExpression)',
+	'>',
+	'AssignmentPattern',
```

Comment:
Though another child(function.id) can't be `AssignmentPattern`, I still prefer use `AssignmentPattern.params`.